cplusplus/VImage.cpp

```diff
@@ -169,7 +169,7 @@ VOption::set( const char *name, const char *value )
 
 // input image
 VOption *
-VOption::set( const char *name, VImage value )
+VOption::set( const char *name, const VImage &value )
```

Comment:
Sorry to return to this old change.

I guess this is for efficiency, is that right? By design, `VInterpolate` is just a pointer (`sizeof(VImage) == 8` etc.) so I don't think there's any saving for switching to call-by-reference, and it adds complexity.
 
I'm inclined to revert this, unless there is a good reason for call by reference.

Reply:
It's intended for efficiency. Indeed it makes no sense here and I'm also not sure whether it's ABI-compatible.

Feel free to revert this. Perhaps we should also pass-by-value within the `new_from_stream` and `write_to_stream` functions?:
https://github.com/libvips/libvips/pull/1443/files#diff-812eb578264cc528d97b8955370e75aaR517
https://github.com/libvips/libvips/pull/1443/files#diff-812eb578264cc528d97b8955370e75aaR573

Reply:
Hiya, yes, I flipped them all over to call-by-value in https://github.com/libvips/libvips/pull/1443/commits/123c52be7041f8d1ea9e937643b3a0df317b02ea

Reply:
Yes, it would break ABI: `type&` parameters have extra code at the caller.

Though C++ ABI breaks are very hard to avoid, so it  probably doesn't matter.

Reply:
Thanks! fwiw, the reason I switched them to a const reference was due to this check by `clang-tidy`:
https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-value-param.html

Somehow it detects that `VImage` is expensive to copy.

Reply:
Huh strange. There is a copy constructor that is triggered to adjust the ref counts, perhaps that's enough to make the warning happen?

Reply:
It seems that when I remove the VObject destructor (`~VObject()`) and copy constructor
(`VObject( const VObject &a ) : vobject( a.vobject )`) the warnings disappear, so I think you're right.